### PR TITLE
Improve Area fault tolerance, add TCAV support, and expand Beampath aggregation APIs

### PR DESCRIPTION
**Pull request by kabanaty**
## Summary
This PR improves Area and Beampath reliability, adds TCAV support, and expands Beampath device query APIs.  Addresses #4 .

## Changes
- Added per-device pruning in Area validation so invalid devices are skipped instead of failing the whole Area.
- Added filtering of unsupported top-level device keys in reader Area construction.
- Added TCAV support in Area and reader, including TCAV collection handling.
- Added Area representation with total device count, per-type counts, and populated types.
- Added Beampath representation with instantiated areas and device summary.
- Added Beampath cross-area helpers:
  - find_device by name
  - get_all_magnets, get_all_screens, get_all_wires, get_all_bpms, get_all_lblms, get_all_pmts, get_all_tcavs, get_all_devices
- Added Beampath convenience properties by device type:
  - magnets, screens, wires, bpms, lblms, pmts, tcavs, devices

## Tests
- Added new Beampath test suite for repr, aggregation, lookup, and properties.
- Added/updated Area and reader tests for tolerant creation and TCAV support.
- Full suite passes: 101 passed, 25 subtests passed.

## Impact
- Area/Beampath creation is more resilient to partial bad config.
- TCAVs are now modeled and exposed through Area and Beampath APIs.
- Beampath inspection and cross-area device access are much easier.

### CODE REVIEW

File: slac_devices/beampath.py [hunk 1]
```diff
@@ -73,3 +81,198 @@ def contains_areas(
...
+            f"num_areas={len(self.areas)}, "
+            f"area_names={area_names}, "
+            f"total_devices={sum(counts.values())}, "
+            f"counts={counts}, "
+            f"populated_types={populated_types})"
+        )
+
+    def find_device(
+        self, device_name: str
+    ) -> Optional[Tuple[str, str, Union[Magnet, Screen, Wire, BPM, LBLM, PMT, TCAV]]]:
```
Comment (shamin-slac) [inline slac_devices/beampath.py]: Does Device not work here instead of the Union?

File: slac_devices/tcav.py [hunk 2]
```diff
@@ -244,3 +247,26 @@ def l_eff(self, length):
...
+
+    @field_validator("tcavs", mode="before")
+    def validate_tcavs(cls, v) -> Dict[str, TCAV]:
+        for name, tcav in v.items():
+            tcav = dict(tcav)
+            tcav.update({"name": name})
+            v.update({name: tcav})
+        return v
+
+    def _make_tcav_names_list_from_args(
```
Comment (shamin-slac) [inline slac_devices/tcav.py]: Where is this used?

File: slac_devices/area.py [hunk 3]
```diff
@@ -111,59 +163,146 @@ def __init__(
...
+        counts = {}
+        for device_type, (collection, attr_name) in collection_map.items():
+            if collection is None:
+                counts[device_type] = 0
+                continue
+            devices = getattr(collection, attr_name, None) or {}
+            counts[device_type] = len(devices)
+        return counts
+
+    def __repr__(self) -> str:
```
Comment (shamin-slac) [inline slac_devices/area.py:186]: Where are we using these string representations?

File: tests/test_beampath.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+    def test_beampath_repr_empty_areas(self):
+        """Test __repr__ for beampath with no areas."""
+        bp = Beampath.model_construct(name="TEST_BP", areas=None)
+        repr_str = repr(bp)
+        self.assertIn("TEST_BP", repr_str)
+        self.assertIn("areas=[]", repr_str)
+
+    def test_beampath_repr_with_areas(self):
+        """Test __repr__ shows area info and device counts."""
+        # Create minimal area with one magnet
```
Comment (shamin-slac) [inline tests/test_beampath.py]: It looks like you were able to instantiate the area with a magnet collection without magnets. This seems to give the attributes you're testing for in the string, so I don't know if you wanted to include a magnet or if it makes a difference.

File: tests/test_beampath.py [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+        result = bp.find_device("S1")
+        self.assertIsNotNone(result)
+        self.assertEqual(result[0], "AREA2")
+        self.assertEqual(result[1], "screens")
+
+    def test_find_device_no_configured_areas(self):
+        """Test find_device with unconfigured beampath."""
+        bp = Beampath.model_construct(name="UNCONFIGURED", areas=None)
+        result = bp.find_device("SOMETHING")
+        self.assertIsNone(result)
```
Comment (shamin-slac) [inline tests/test_beampath.py:155]: Should an unconfigured beampath be caught upon beampath instantiation? If you try to find device and get None, this could either be because the beampath is not configured or because the device is not in that beampath.

File: tests/test_reader.py [hunk 6]
```diff
@@ -60,3 +61,110 @@ def test_create_magnet_with_config_and_name_creates_one_magnet(self):
...
+                            "rf_freq": 2856.0,
+                        },
+                    }
+                }
+            }
+            result = create_area(area="LI30")
+
+        self.assertIsInstance(result, Area)
+        self.assertIsNone(result.bpms)
+        self.assertIsNone(result.magnets)
```
Comment (shamin-slac) [inline tests/test_reader.py]: I'm not sure what these `assertIsNone` checks are doing. If the device type was not inputted properly, wouldn't this be caught by the `assertIsNotNone` check below?

**Review by shamin-slac — COMMENTED** [review]
Looks good. This will definitely make the beampaths and areas more reliable. I just had some minor questions.

File: slac_devices/beampath.py (hunk 1 above)
Comment (kabanaty) [reply slac_devices/beampath.py]: Device will absolutely work better here than a massive Union.  Same for the `devices` property.

File: slac_devices/area.py [hunk 7]
```diff
@@ -29,8 +29,51 @@
...
+            print(
+                f"Skipping {device_type} {name} in {area_name}: "
+                + "device definition is not a dictionary."
+            )
+            continue
+
+        payload_copy = dict(payload)
+        payload_copy.pop("name", None)
+        try:
+            device_class(name=name, **payload_copy)
```
Comment (eloise-nebula) [inline slac_devices/area.py]: Could we do `payload_copy["name"] = name` such we can call `device_class(**payload_copy)`?

File: slac_devices/tcav.py (hunk 2 above)
Comment (kabanaty) [reply slac_devices/tcav.py]: Great catch, it doesn't appear to be used anywhere. I can remove it.

File: slac_devices/area.py (hunk 3 above)
Comment (kabanaty) [reply slac_devices/area.py:186]: They're not necessarily 'used'.  If you call an `Area` or `Beampath` from the shell, the string representation is a massive list of every `Device` in a given `Area`/`Beampath` including every PV object and connection info for each device. This is a QOL enhancement for this use case.

**Review by eloise-nebula — COMMENTED** [review]
It looks like this code works, but I think I might have solved this problem in a different way. Check out this from my slac-db latest PR https://github.com/slaclab/slac-db/pull/37:

```
def _build_devices(area, device_type):
    devices = slac_db.device.get_devices(
        area=area, device_type=device_type
    )
    for d in devices:
        if device_type == "INST" and not d.startswith("PMT"):
            continue
        if device_type == "LCAV":
            r = slac_db.oracle.get_device_row(d)
            if r["engineering name"] != "TRANS_DEFL":
                continue
        cs = _build_controls_information(d)
        meta = _build_metadata(area, device_type, d)
        yield d, {
            "controls_information": cs,
            "metadata": meta,
        }
```

We can rebuild YAML files by area and device type with massively reduced boilerplate. It's already been tested against the current YAML file base. I think it solves a lot of problems this PR is trying to solve.

File: tests/test_beampath.py (hunk 4 above)
Comment (kabanaty) [reply tests/test_beampath.py]: I don't think it makes a difference for this test. I'll update the docstring for clarity.

File: tests/test_reader.py (hunk 6 above)
Comment (kabanaty) [reply tests/test_reader.py]: Yeah these are pointless. I'll remove them.

File: tests/test_beampath.py (hunk 5 above)
Comment (kabanaty) [reply tests/test_beampath.py:155]: This is testing that find_device doesn't blow up when there are no configured areas and instead just returns `None`.  This test is less about the `"UNCONFIGURED" Beampath` and more about `.find_device()` handling an edge case.  `model_construct` bypasses Pydantic validation, so it looks weird, but I can rewrite this.

File: slac_devices/area.py (hunk 7 above)
Comment (kabanaty) [reply slac_devices/area.py]: Sure, I've made that change
